Source/WebCore/platform/mediastream/gstreamer/GStreamerMediaStreamSource.cpp

```diff
@@ -309,7 +309,7 @@ class InternalSource final : public MediaStreamTrackPrivate::Observer,
 
         if (m_track.enabled()) {
             GST_TRACE_OBJECT(m_src.get(), "Pushing video frame from enabled track");
-            pushSample(gstSample);
+            pushSample(WTFMove(gstSample));
```

Comment:
Just this call site or all of these as well?
https://github.com/WebKit/WebKit/blob/c05bfb093f5d4cb63fd341003b02b7d677662879/Source/WebCore/platform/mediastream/gstreamer/GStreamerMediaStreamSource.cpp#L328

https://github.com/WebKit/WebKit/blob/c05bfb093f5d4cb63fd341003b02b7d677662879/Source/WebCore/platform/mediastream/gstreamer/GStreamerMediaStreamSource.cpp#L337

https://github.com/WebKit/WebKit/blob/c05bfb093f5d4cb63fd341003b02b7d677662879/Source/WebCore/platform/mediastream/gstreamer/GStreamerMediaStreamSource.cpp#L370

Reply:
I don't think this is correct, the sample is owned by the `VideoFrame`, if you move the sample here and there's more than one observer registered, things will go bad...

In which context exactly did you notice a leak?

Reply:
I'm thinking this may have actually been related to a [gstreamer bug](https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/2928) due to parent buffer metadata refs leaking down the pipeline even when the parent buffers were not being referenced(which could prevent the memory mapped parent buffers from being recycled by pipewire).